src/swtpm/swtpm_nvfile.c

```diff
@@ -557,6 +569,39 @@ SWTPM_NVRAM_StoreData_Intern(const unsigned char *data,
         }
     }
 
+    if (rc == 0 && fd >= 0) {
+        TPM_DEBUG(" SWTPM_NVRAM_StoreData: Opening dir %s\n", state_directory);
+        dir_fd = open(state_directory, O_RDONLY);
```

Comment:
But why do we need to sync on the whole directory if we are only writing a file? Can you give me a pointer to read why this would be necessary?

Reply:
Sure.
Man fsync(2): https://man7.org/linux/man-pages/man2/fdatasync.2.html

> Calling fsync() does not necessarily ensure that the entry in the
> directory containing the file has also reached disk.  For that an
> explicit fsync() on a file descriptor for the directory is also
> needed.

Reply:
Can you add a comment to the code that indicates that this is necessary, please.

Reply:
Sure, done.